lib/mnlists/mnListDiff.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+var BufferReader = require('../encoding/bufferreader');
+var BufferWriter = require('../encoding/bufferwriter');
+var BufferUtil = require('../util/buffer');
+var _ = require('lodash');
+var $ = require('../util/preconditions');
+
+/* @param {*} - A Buffer, JSON string, or Object representing a MerkleBlock
+* @returns {MerkleBlock}
+* @constructor
+*/
+function MnListDiff(arg) {
+
+  if (!(this instanceof MnListDiff)) {
+    return new MnListDiff(arg);
+  }
+
+  var info = {};
+  if (BufferUtil.isBuffer(arg)) {
+    info = MerkleBlock._fromBufferReader(BufferReader(arg));
+  } else if (_.isObject(arg)) {
+    info = {
+      baseBlockHash: arg.baseBlockHash,
+      blockHash: arg.blockHash,
+      totalTransactions: arg.totalTransactions,
+      merkleHashes: arg.merkleHashes,
+      merkleFlags: arg.merkleFlags,
+      cbTx: arg.cbTx,
+      deletedMNs: arg.deletedMNs,
+      mnList: arg.mnList,
+    }
+  }
+  else {
+    throw new TypeError('Unrecognized argument for MerkleBlock');
+  }
+  _.extend(this, info);
+  this._flagBitsUsed = 0;
+  this._hashesUsed = 0;
+  return this;
+}
+
+MnListDiff._fromBufferReader = function _fromBufferReader(br) {
+  $.checkState(!br.finished(), 'No mndifflist data received');
+  var info = {};
+  info.baseBlockHash = br.read();
```

Comment:
You always need to specify a length when reading a sequence of bytes into a variable. It will throw an error here

Reply:
thanks will correct.  what about thing like `cbTx` with **variable** size?